src/test/java/io/stargate/sgv2/jsonapi/api/v1/IndexingConfigIntegrationTest.java

```diff
@@ -0,0 +1,893 @@
+                  }
+                }
+              }
+                    """;
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(filterData2)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, allowManyIndexingCollection)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("data", is(nullValue()))
+          .body("errors[0].message", endsWith("The filter path ('address.street') is not indexed"))
+          .body("errors[0].errorCode", is("UNINDEXED_FILTER_PATH"))
+          .body("errors[0].exceptionClass", is("JsonApiException"));
+    }
+
+    @Test
+    public void incrementalPathInArray() {
+      // explicitly allow "name" "address.city", implicitly deny "_id" "address.street"
+      // String and array in array - no incremental path, the path is "address" - should be allowed
+      // but no data return
+      String filterData1 =
+          """
+                  {
+                    "find": {
+                      "filter": {
+                        "address": {
+                          "$in": [[{"street": "1 banana street"}], "abc", ["def"]]
+                        }
+                      }
+                    }
+                  }
+                  """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(filterData1)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, allowManyIndexingCollection)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("data", is(nullValue()))
+          .body("errors[0].message", endsWith("The filter path ('address.street') is not indexed"))
+          .body("errors[0].errorCode", is("UNINDEXED_FILTER_PATH"))
+          .body("errors[0].exceptionClass", is("JsonApiException"));
+      // explicitly allow "name" "address.city", implicitly deny "_id" "address.street"
+      String filterData2 =
+          """
+                      {
+                        "find": {
+                          "filter": {
+                            "address": {
+                              "$in": [[{"city": "monkey town"}]]
+                            }
+                          }
+                        }
+                      }
+                      """;
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(filterData2)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, allowManyIndexingCollection)
+          .then()
+          .statusCode(200)
+          .body("status", is(nullValue()))
+          .body("errors", is(nullValue()))
+          .body("data.documents", hasSize(1));
```

Comment:
@Yuqi-Du Is this use case of $in allowed? I didn't get any errors and get any data

Reply:
Yes, this is allowed. Weird use case though.
You should have data return back.

Reply:
If you insert data like this, 
`{
    "insertOne": {
        "document": {
                    "address": [
                {
                    "city": "monkey town"
                }
            ]
} }`
I do not think deny or allow should work, since we actually do not have address.city field here.

Reply:
No, I don't think that is legal `filter`, wrt nesting. At main level there is always (full) path as key, and then JSON Object with operators as keys and so on.

Reply:
My bad, I found a bug in my validation logic. Thank you Yuqi!